DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerMaskedTagRule.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Rules
+{
+    public class AnonymizerMaskedTagRule : AnonymizerRule
+    {
+        public AnonymizerMaskedTagRule(DicomMaskedTag maskedTag, string method, JObject ruleSetting, string description, IAnonymizerProcessorFactory processorFactory = null, IAnonymizerSettingsFactory settingsFactory = null)
+            : base(method, description, ruleSetting, processorFactory, settingsFactory)
+        {
+            EnsureArg.IsNotNull(maskedTag, nameof(maskedTag));
+
+            MaskedTag = maskedTag;
+        }
+
+        public DicomMaskedTag MaskedTag { get; set; }
+
+        public override List<DicomItem> LocateDicomTag(DicomDataset dataset, ProcessContext context)
```

Comment:
Looks like you can have a common LocateDicomTag() method in the base class. And you can override a method simply to tell which Item should be included. public abstract bool IsTargetItem(DicomItem item)

Reply:
Could refine later.